### DISCUSSION TITLE
How to include images in the web page sent to WebUI?

### DISCUSSION THREAD

**Discussion by dcipher** [webui-dev/webui#510] [Q&A] [answered]
Hi

Is it possible to just include the path to an image or does the image need to be converted to base64 before it can be included in the web page sent to WebUI.  I have tried different paths but the image is not displayed so I am assuming it must be included in the web page as base64 encoded text.  Is that the only way to include images?

Thanks,
Simon

**Comment by AlbertShown** [comment, marked as answer]
webui works like any normal web server, so you can do it in any ways you want, file path, base64, or even binary embedded in the compiled program.

If file path did not work, it mutely because you did not setup the root path.
Documentation: https://webui.me/docs/2.5/#/?id=set_root_folder

**Comment by AlbertShown** [comment]
See this example: https://github.com/webui-dev/webui/tree/main/examples/C/text-editor
